language/move-vm/types/src/loaded_data/types.rs

```diff
@@ -132,6 +134,18 @@ impl Type {
 
         Ok(res)
     }
+
+    pub fn is_resource(&self) -> VMResult<bool> {
+        use Type::*;
+
+        match self {
+            Bool | U8 | U64 | U128 | Address | Reference(_) | MutableReference(_) => Ok(false),
+            Vector(ty) => ty.is_resource(),
+            Struct(struct_ty) => Ok(struct_ty.is_resource),
+            TyParam(_) => Err(VMStatus::new(StatusCode::UNKNOWN_INVARIANT_VIOLATION_ERROR)
```

Comment:
Should we extend `is_resource`  with `type_arguments: &[Type]` so we can implement this?

Reply:
Technically we can but I don't think it's useful. The reason is that with the latest type system refactor, type resolution now *requires* type arguments, meaning all types returned are fully instantiated. 

The `TyParam` only comes into play during resolution -- when loading a struct definition, we create one with uninstantiated type parameters, cache it and then perform substitution with the type arguments specified by the caller. This means it should never escape the cache.

Reply:
Seems good to me!
Maybe add an invariant violation for doing some operation on an uninstantiated type? Or at least a comment :)